pkg/storage/stores/indexshipper/gatewayclient/gateway_client.go

```diff
@@ -142,6 +146,20 @@ func NewGatewayClient(cfg IndexGatewayClientConfig, r prometheus.Registerer, log
 			return igPool, nil
 		}
 
+		// Replication factor plus additional room for JOINING/LEAVING instances
+		// See also ring.GetBufferSize
+		bufSize := cfg.Ring.ReplicationFactor() * 3 / 2
```

Comment:
Could you use a constant with the same doc string?

Reply:
You don't want this to be a constant, because the RF is a setting that can change.